driver-core/src/test/java/com/datastax/driver/core/IndexMetadataTest.java

```diff
@@ -48,192 +52,304 @@
-        assertThat(column).hasSingleIndex(index);
-        assertThat(column).isSameAs(index.getColumn("list_full"));
+        assertThat((TableMetadata) column.getParent()).hasIndex(index);
     }
 
     @Test(groups = "short")
     @CassandraVersion(major = 2.2)
-    public void should_flag_map_index_type_as_entries() {
-        String createEntriesIndex = String.format("CREATE INDEX map_entries_index ON %s.indexing (ENTRIES(map_entries));", keyspace);
+    public void should_create_metadata_for_index_on_map_entries() {
+        String createEntriesIndex = String.format("CREATE INDEX map_entries_index ON %s.indexing (entries(map_entries));", keyspace);
         session.execute(createEntriesIndex);
         ColumnMetadata column = getColumn("map_entries");
         IndexMetadata index = getIndex("map_entries_index");
-        assertThat(index).hasName("map_entries_index")
-            .isNotKeys()
-            .isNotFull()
-            .isEntries()
+        assertThat(index)
+            .hasName("map_entries_index")
+            .hasParent((TableMetadata)column.getParent())
             .isNotCustomIndex()
-            .hasSingleColumn(getColumn("map_entries"))
+            .hasTarget("entries(map_entries)")
+            .hasKind(COMPOSITES)
             .asCqlQuery(createEntriesIndex);
-        assertThat(index).isSameAs(column.getIndex("map_entries_index"));
-        assertThat(column).hasSingleIndex(index);
-        assertThat(column).isSameAs(index.getColumn("map_entries"));
+        assertThat((TableMetadata) column.getParent()).hasIndex(index);
+    }
+
+    @Test(groups = "short")
+    @CassandraVersion(major = 3.0)
+    public void should_allow_multiple_indexes_on_map_column() {
+        String createEntriesIndex = String.format("CREATE INDEX map_all_entries_index ON %s.indexing (entries(map_all));", keyspace);
+        session.execute(createEntriesIndex);
+        String createKeysIndex = String.format("CREATE INDEX map_all_keys_index ON %s.indexing (keys(map_all));", keyspace);
+        session.execute(createKeysIndex);
+        String createValuesIndex = String.format("CREATE INDEX map_all_values_index ON %s.indexing (values(map_all));", keyspace);
+        session.execute(createValuesIndex);
+
+        ColumnMetadata column = getColumn("map_all");
+        TableMetadata table = (TableMetadata)column.getParent();
+
+        assertThat(getIndex("map_all_entries_index"))
+            .hasParent(table)
+            .asCqlQuery(createEntriesIndex);
+
+        assertThat(getIndex("map_all_keys_index"))
+            .hasParent(table)
+            .asCqlQuery(createKeysIndex);
+
+        assertThat(getIndex("map_all_values_index"))
+            .hasParent(table)
+            .asCqlQuery(createValuesIndex);
+    }
+
+    @Test(groups = "short", enabled = false)
+    @CassandraVersion(major = 3.0)
+    public void should_create_metadata_for_index_on_materialized_view() {
```

Comment:
It'd probably be fine to just delete this test and if support gets added back we can readd it then